src/GlobalErrorHandler.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import * as Sentry from '@sentry/browser'
+import GenericError from './components/GenericError'
+// import { DAONotFound } from './errors'
+import env from './environment'
+import GlobalErrorScreen from './components/GlobalErrorScreen'
+import { getNetworkName } from './lib/web3-utils'
+
+const SENTRY_DSN = env('SENTRY_DNS')
+
+class GlobalErrorHandler extends React.Component {
+  static propTypes = {
+    children: PropTypes.node,
+  }
+
+  state = { error: null, errorStack: null }
+
+  componentDidCatch(error, errorInfo) {
+    this.setState({
+      error,
+      errorStack: errorInfo.componentStack
+        .replace(/^\n+|\n+$/g, '')
+        .replace(/^ {4}/gm, ''),
+    })
+
+    // Once this point is reached, the app can not recover because the routing
+    // system, being below this component in the tree, is not functional
+    // anymore. To make hash changes work despite this (e.g. by pressing the
+    // back button in the browser), the page need to be reloaded.
+    window.removeEventListener('hashchange', this.handleHashchange)
+    window.addEventListener('hashchange', this.handleHashchange)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('hashchange', this.handleHashchange)
+  }
+
+  handleReportClick = () => {
```

Comment:
Let's just always report new errors caught here to sentry, and use the button to show the report dialog.